File: src/main/java/com/waridley/ttv/logger/backend/local/FileMessageLogger.java
package com.waridley.ttv.logger.backend.local;

import com.github.twitch4j.chat.events.channel.ChannelMessageEvent;
import com.waridley.ttv.logger.ChatLogger;
import lombok.extern.slf4j.Slf4j;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.PrintWriter;

@Slf4j
public class FileMessageLogger implements ChatLogger.MessageLogger {
	private PrintWriter logWriter;
	
	public FileMessageLogger(String filePath) throws IOException {
		File file = new File(filePath);
		if(file.getParentFile().mkdirs()) log.info("Created parent directories for log file");
		logWriter = new PrintWriter(new FileOutputStream(file, true));
		log.info("Logging chat to file: {}", filePath);
	}
	
	@Override
	public void logMessage(ChannelMessageEvent event) {
		logWriter.printf("[%s] [%s] [%s] %s\n",
				event.getFiredAt().toInstant().toString(),
				event.getChannel().getName(),
				event.getUser().getName(),
				event.getMessage());
		logWriter.flush();
	}
}

File: src/main/java/com/waridley/ttv/logger/backend/local/FileMessageLogger.kt
package com.waridley.ttv.logger.backend.local

import com.github.twitch4j.chat.events.channel.ChannelMessageEvent
import com.waridley.ttv.logger.ChatLogger.MessageLogger
import org.slf4j.LoggerFactory
import java.io.File
import java.io.FileOutputStream
import java.io.PrintWriter

class FileMessageLogger(filePath: String) : MessageLogger {
	private val logWriter: PrintWriter
	override fun logMessage(event: ChannelMessageEvent) {
		logWriter.printf("[%s] [%s] [%s] %s\n",
				event.firedAt.toInstant().toString(),
				event.channel.name,
				event.user.name,
				event.message)
		logWriter.flush()
	}
	
	companion object {
		private val log = LoggerFactory.getLogger(FileMessageLogger::class.java)
	}
	
	init {
		val file = File(filePath)
		if (file.parentFile?.mkdirs() == true) log.info("Created parent directories for log file")
		logWriter = PrintWriter(FileOutputStream(file, true))
		log.info("Logging chat to file: {}", filePath)
	}
}